### PR TITLE
Music Album Data Persistence

### LINKED ISSUE #26: [2.5] Data persistence for the MusicAlbum class
All data should be preserved by saving collections in `.json` files.

### PR DESCRIPTION
**Pull request by rpire**
## Changes made:
- Created the `storage.rb` file introducing the `Storage` module.
- Created the `local` folder to store data in JSON format.
- Refactored the `App` and `MusicAlbum` classes to work alongside `Storage` and implement data persistence.

closes #26

### CODE REVIEW

**Review by aliabbani — APPROVED** [review]
Hi @rpire , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🥇 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

File: storage.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+module Storage
+  def load_albums
+    json_albums = File.read('./local/music_album.json')
+    albums_data = JSON.parse(json_albums)
+    albums_data.each do |element|
+      album = MusicAlbum.new(element['name'], element['publish_date'], element['on_spotify'])
+      full_item = [album, element['genre']]
+      @music_albums << album
+      @full_items << full_item
```
Comment (medaminedev66) [inline storage.rb:11]: Here, @rpire. We can push the objects directly to @music_albums and @full_items to keep it simple and dry.

File: storage.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    end
+  end
+
+  def load_genres
+    json_genres = File.read('./local/genre.json')
+    genres_data = JSON.parse(json_genres)
+    genres_data.each do |element|
+      genre = Genre.new(element['name'])
+      @genres << genre
```
Comment (medaminedev66) [inline storage.rb:20]: The same thing here. Try to push the object directly. to @genre array.

File: storage.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    end
+    File.write('./local/music_album.json', JSON.dump(albums_arr))
+  end
+
+  def save_genres
+    File.new('./local/genre.json', 'a') unless File.exist?('./local/genre.json')
+    genres_arr = []
+    @genres.each do |genre|
+      stored_genre = { name: genre.name }
+      genres_arr << stored_genre
```
Comment (medaminedev66) [inline storage.rb:58]: The same thing here. Try to push the object directly. to @stored_genre array.

File: storage.rb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    File.new('./local/music_album.json', 'a') unless File.exist?('./local/music_album.json')
+    albums_arr = []
+    @music_albums.each do |album|
+      stored_album = {
+        name: album.name,
+        publish_date: album.publish_date,
+        on_spotify: album.on_spotify,
+        genre: album.genre.name
+      }
+      albums_arr << stored_album
```
Comment (medaminedev66) [inline storage.rb:48]: The same thing here. Try to push the object directly. to @albums_arr array. To keep it simple and dry.

**Pull request merged**